src/overlays/actors/ovl_Bg_Dkjail_Ivy/z_bg_dkjail_ivy.c

```diff
@@ -28,47 +36,176 @@ const ActorInit Bg_Dkjail_Ivy_InitVars = {
+    Vec3f spC8;
+    Vec3f spBC;
+    Vec3f spB0;
+    s16 i2;
+
+    Matrix_RotateYS(this->dyna.actor.home.rot.y, MTXMODE_NEW);
+
+    for (i = 0, i2 = 0; i < 50; i++, i2 += 0x4e20) {
+        spD4.y = (i * (90.0f / 49.0f)) + 2.0f;
+        if (spD4.y < 45.0f) {
+            phi_fs0 = ((1.0f / 180.0f) * spD4.y) + 0.75f;
+        } else if (spD4.y < 90.0f) {
+            phi_fs0 = ((-1.0f / 180.0f) * spD4.y) + 1.25f;
+        } else {
+            phi_fs0 = ((-1.0f / 30.0f) * spD4.y) + 3.75f;
+        }
+
+        spD4.x = Math_SinS(i2) * 40.0f * phi_fs0;
+        spD4.z = (Rand_ZeroOne() * 6.0f) - 3.0f;
+
+        Matrix_MultVec3f(&spD4, &spC8);
+
+        spBC.x = (Rand_ZeroOne() - 0.5f) + (spC8.x * 0.075f);
+        spBC.y = 2.0f * Rand_ZeroOne();
+        spBC.z = (Rand_ZeroOne() - 0.5f) + (spC8.z * 0.075f);
+
+        spC8.x += this->dyna.actor.world.pos.x;
+        spC8.y += this->dyna.actor.world.pos.y;
+        spC8.z += this->dyna.actor.world.pos.z;
+
+        EffectSsKakera_Spawn(play, &spC8, &spBC, &spC8, -0x82, 0x40, 0x28, 0, 0, sLeafScales[i & 3], 0, 0, 44, -1,
+                             GAMEPLAY_KEEP, sLeafDlists[(s32)Rand_Next() > 0]);
+
+        if ((i > 20) && ((i & 1) != 0)) {
+            spB0.x = (Rand_ZeroOne() - 0.5f) * 0.2f;
+            spB0.y = (Rand_ZeroOne() * 0.02f) - 0.1f;
+            spB0.z = (Rand_ZeroOne() - 0.5f) * 0.2f;
+
+            func_800B12F0(play, &spC8, &gZeroVec3f, &spB0, ((Rand_Next() >> 26) + 10), (Rand_Next() >> 28), 50);
+        }
+    }
+}
+
+void BgDkjailIvy_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    BgDkjailIvy* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 0);
+    Collider_InitCylinder(play, &this->collider);
+
+    if (Flags_GetSwitch(play, BG_DKJAIL_GET_SWITCH(thisx))) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        DynaPolyActor_LoadMesh(play, &this->dyna, &object_dkjail_obj_Colheader_0011A8);
+        Collider_SetCylinder(play, &this->collider, &this->dyna.actor, &sCylinderInit);
+        Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+        this->alpha = 255;
+        func_80ADE6AC(this);
+    }
+}
+
+void BgDkjailIvy_Destroy(Actor* thisx, PlayState* play) {
+    BgDkjailIvy* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void func_80ADE6AC(BgDkjailIvy* this) {
+    this->actionFunc = BgDkjailIvy_SetCollision;
+}
+
+void BgDkjailIvy_SetCollision(BgDkjailIvy* this, PlayState* play) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        this->dyna.actor.flags |= ACTOR_FLAG_10;
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+        func_80ADE734(this);
+    } else {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    }
+}
+
+void func_80ADE734(BgDkjailIvy* this) {
```

Comment:
```suggestion
void BgDkjailIvy_SetupCutscene(BgDkjailIvy* this) {
```
This function is the one that should be called SetupCutscene. The other one would be more like `BeginCutscene`.